decode_test.go

```diff
@@ -1241,8 +1260,8 @@ func TestUnmarshalerWholeDocument(t *testing.T) {
 }
 
 func TestUnmarshalerTypeError(t *testing.T) {
-	unmarshalerResult[2] = &yaml.TypeError{[]string{"foo"}}
-	unmarshalerResult[4] = &yaml.TypeError{[]string{"bar"}}
+	unmarshalerResult[2] = &yaml.TypeError{[]*yaml.UnmarshalError{{errors.New("foo"), 1, 1}}}
+	unmarshalerResult[4] = &yaml.TypeError{[]*yaml.UnmarshalError{{errors.New("bar"), 1, 1}}}
```

Comment:
Please use field names, it could break things if there is a refactoring or simply if fields are swapped.


```suggestion
	unmarshalerResult[2] = &yaml.TypeError{[]*yaml.UnmarshalError{{Err: errors.New("foo"), Line: 1, Column: 1}}}
	unmarshalerResult[4] = &yaml.TypeError{[]*yaml.UnmarshalError{{Err: errors.New("bar"), Line: 1, Column: 1}}}
```